src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c

```diff
@@ -480,50 +521,71 @@ static int hwmon_sdm_probe(struct platform_device *pdev)
 		goto failed;
 	}
 
+	sdm->cache_expire_secs = HWMON_SDM_DEFAULT_EXPIRE_SECS;
+
 	xocl_info(&pdev->dev, "hwmon_sdm driver probe is successful");
-	return 0;
+	return err;
 
 failed:
 	hwmon_sdm_remove(pdev);
 	return err;
 }
 
-static void hwmon_sdm_get_sensors_list(struct platform_device *pdev)
+static int hwmon_sdm_update_sensors_by_type(struct platform_device *pdev,
+                                            enum xgq_sdr_repo_type repo_type,
+                                            bool create_sysfs)
 {
 	struct xocl_hwmon_sdm *sdm = platform_get_drvdata(pdev);
 	xdev_handle_t xdev = xocl_get_xdev(pdev);
-	uint32_t resp_size;
 	int repo_id, ret;
-
 	//TODO: request vmc/vmr for bdinfo resp size
-	resp_size = 4 * 1024;
-	repo_id = sdr_get_id(SDR_TYPE_BDINFO);
-	sdm->sensor_data[repo_id] = (char*)kzalloc(sizeof(char) * resp_size,
-                                               GFP_KERNEL);
-	ret = xocl_xgq_collect_bdinfo_sensors(xdev, sdm->sensor_data[repo_id],
-                                          resp_size);
-	if (!ret) {
-		ret = parse_sdr_info(sdm->sensor_data[repo_id], sdm, false);
-		if (!ret)
-			sdm->sensor_data_avail[repo_id] = true;
-	} else {
-		xocl_err(&pdev->dev, "request is failed with err: %d", ret);
+	uint32_t resp_size = 4 * 1024;
+
+	repo_id = sdr_get_id(repo_type);
+	if (repo_id < 0) {
+		xocl_err(&pdev->dev, "received invalid sdr repo type: %d", repo_type);
+		return -EINVAL;
 	}
 
-	//TODO: request vmc/vmr for temp resp size
-	resp_size = 4 * 1024;
-	repo_id = sdr_get_id(SDR_TYPE_TEMP);
 	sdm->sensor_data[repo_id] = (char*)kzalloc(sizeof(char) * resp_size,
                                                GFP_KERNEL);
-	ret = xocl_xgq_collect_temp_sensors(xdev, sdm->sensor_data[repo_id],
-                                        resp_size);
+	ret = xocl_xgq_collect_sensors_by_id(xdev, sdm->sensor_data[repo_id],
+                                         repo_id, resp_size);
 	if (!ret) {
-		ret = parse_sdr_info(sdm->sensor_data[repo_id], sdm, true);
+		ret = parse_sdr_info(sdm->sensor_data[repo_id], sdm, create_sysfs);
 		if (!ret)
 			sdm->sensor_data_avail[repo_id] = true;
 	} else {
 		xocl_err(&pdev->dev, "request is failed with err: %d", ret);
+		sdm->sensor_data_avail[repo_id] = false;
+	}
+
+	return ret;
+}
+
+static void hwmon_sdm_get_sensors_list(struct platform_device *pdev, bool create_sysfs)
+{
+	(void) hwmon_sdm_update_sensors_by_type(pdev, SDR_TYPE_BDINFO, false);
```

Comment:
We need to pass create_sysfs instead of false?

Reply:
we are not creating any sysfs entries for bdinfo. So, passing false for BDINFO request.

Reply:
Thanks Raj for clarifying. This means, we need to capture bdinfo in some other way sothat xbutil can read.

Reply:
yes, plan is to create a sysfs entries under hwmon_sdm device, so that xbutil can read these values.